toolchain/check/handle_namespace.cpp

```diff
@@ -22,6 +23,9 @@ auto HandleParseNode(Context& context, Parse::NamespaceStartId /*node_id*/)
   // Optional modifiers and the name follow.
   context.decl_introducer_state_stack().Push<Lex::TokenKind::Namespace>();
   context.decl_name_stack().PushScopeAndStartName();
+  // Namespaces can't be generic, but we might have parsed a generic parameter
+  // in their name, so enter a generic scope just in case.
+  StartGenericDecl(context);
```

Comment:
Minor, but it feels weird to me how the push and pop is in a different order, versus having `StartGenericDecl` called before `PushScope` (in all of these)

Reply:
Changed throughout. The thing that seemed to make the most sense was to enter the generic scope first, because the whole of the generic declaration is supposed to be in that scope, including the various other things that we push.